### PR TITLE
Add Sample Code for HIPBLASLT_EPILOGUE_GELU_AUX_BIAS and HIPBLASLT_EPILOGUE_DGELU_BGRAD

### PR DESCRIPTION
**Pull request by waynehuangntu**
**Title:**
Add Sample Code for HIPBLASLT_EPILOGUE_GELU_AUX_BIAS and HIPBLASLT_EPILOGUE_DGELU_BGRAD

**Description:**
This PR introduces sample code demonstrating the usage of HIPBLASLT_EPILOGUE_GELU_AUX_BIAS and HIPBLASLT_EPILOGUE_DGELU_BGRAD epilogues during matrix multiplication operations in HIPBLASLt. The code illustrates how to apply these epilogues for common scenarios such as forward and backward passes in neural networks, specifically focusing on the GELU (Gaussian Error Linear Unit) activation function.

**Key Highlights:**

* GELU Activation with Bias:
The HIPBLASLT_EPILOGUE_GELU_AUX_BIAS epilogue applies the GELU activation function along with an auxiliary bias to the matrix multiplication result. The provided sample code showcases how this can be used in forward passes, where the output matrix is first multiplied and then transformed by the GELU function with an added bias.

* Backward Pass Using DGELU:
The HIPBLASLT_EPILOGUE_DGELU_BGRAD epilogue is utilized for backward passes, where it applies the derivative of the GELU function during the gradient calculation. The code example demonstrates how to use this epilogue in conjunction with matrix multiplication to compute gradients with respect to the inputs, which is essential for training neural networks.

**Validation of Functionality:**
The sample code includes validation steps to ensure the correctness of both epilogues:

* GELU with Auxiliary Bias:
The effect of the HIPBLASLT_EPILOGUE_GELU_AUX_BIAS epilogue is validated by applying it to the output of a matrix multiplication and checking that the resulting matrix reflects the expected GELU-transformed values, considering the applied bias.

* Backward Pass with DGELU:
The correctness of HIPBLASLT_EPILOGUE_DGELU_BGRAD is confirmed by comparing the gradients computed using this epilogue with manually calculated gradients. This ensures that the derivative of the GELU function is applied correctly during the backpropagation step.

**1. Validation on GELU_AUX_BIAS:**
Command executed:
`./sample_hipblaslt_gemm_gelu_aux_bias`

Example Input/Output:
Running with parameters:
m = 128, n = 128, k = 128
Bias Vector is of size m x 1 and is initialized as [1, 1, ..., 1].

Matrix A before multiplication (128x128), showing top-left 5x5:
1.0000 1.0000 1.0000 1.0000 1.0000
1.0000 1.0000 1.0000 1.0000 1.0000
1.0000 1.0000 1.0000 1.0000 1.0000
1.0000 1.0000 1.0000 1.0000 1.0000
1.0000 1.0000 1.0000 1.0000 1.0000

Matrix B before multiplication (128x128), showing top-left 5x5:
1.0000 1.0000 1.0000 1.0000 1.0000
1.0000 1.0000 1.0000 1.0000 1.0000
1.0000 1.0000 1.0000 1.0000 1.0000
1.0000 1.0000 1.0000 1.0000 1.0000
1.0000 1.0000 1.0000 1.0000 1.0000

Matrix D after multiplication (128x128), showing top-left 5x5:
129.0000 129.0000 129.0000 129.0000 129.0000
129.0000 129.0000 129.0000 129.0000 129.0000
129.0000 129.0000 129.0000 129.0000 129.0000
129.0000 129.0000 129.0000 129.0000 129.0000
129.0000 129.0000 129.0000 129.0000 129.0000

**2. Validation on DGELU_BGRAD:**

Command executed: 
`./sample_hipblaslt_gemm_dgelu_bgrad`
  
Example Input/Output:
Running with parameters:
m = 128, n = 128, k = 128

Matrix A before multiplication (128x128), showing top-left 5x5:
1.0000 1.0000 1.0000 1.0000 1.0000
1.0000 1.0000 1.0000 1.0000 1.0000
1.0000 1.0000 1.0000 1.0000 1.0000
1.0000 1.0000 1.0000 1.0000 1.0000
1.0000 1.0000 1.0000 1.0000 1.0000

Matrix B before multiplication (128x128), showing top-left 5x5:
1.0000 1.0000 1.0000 1.0000 1.0000
1.0000 1.0000 1.0000 1.0000 1.0000
1.0000 1.0000 1.0000 1.0000 1.0000
1.0000 1.0000 1.0000 1.0000 1.0000
1.0000 1.0000 1.0000 1.0000 1.0000

Matrix Aux Buffer(128x128) as input auxiliary buffer, showing top-left 5x5:
1.0000 1.0000 1.0000 1.0000 1.0000
1.0000 1.0000 1.0000 1.0000 1.0000
1.0000 1.0000 1.0000 1.0000 1.0000
1.0000 1.0000 1.0000 1.0000 1.0000
1.0000 1.0000 1.0000 1.0000 1.0000

Matrix D after multiplication (128x128), showing top-left 5x5:
138.6200 138.6200 138.6200 138.6200 138.6200
138.6200 138.6200 138.6200 138.6200 138.6200
138.6200 138.6200 138.6200 138.6200 138.6200
138.6200 138.6200 138.6200 138.6200 138.6200
138.6200 138.6200 138.6200 138.6200 138.6200

### CODE REVIEW

**Review by jichangjichang — DISMISSED** [review]
lgtm, Please @KKyang to review this PR.

**Review by vin-huang — DISMISSED** [review]
LGTM

File: clients/samples/gemm/sample_hipblaslt_gemm_dgelu_bgrad.cpp [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+    int32_t bias_data_type = HIP_R_16F;
+    CHECK_HIPBLASLT_ERROR(hipblasLtMatmulDescSetAttribute(
+        matmul, HIPBLASLT_MATMUL_DESC_BIAS_DATA_TYPE, &bias_data_type, sizeof(bias_data_type)));
+
+    // Allocate and the bias tensor, the bias gradient will be stored in d_bias buffer
+    std::vector<hipblasLtHalf> h_bias(m, 0.0f);
+    void*                      d_bias;
+    CHECK_HIP_ERROR(hipMalloc(&d_bias, m * sizeof(hipblasLtHalf)));
+    CHECK_HIP_ERROR(
+        hipMemcpy(d_bias, h_bias.data(), m * sizeof(hipblasLtHalf), hipMemcpyHostToDevice));
```
Comment (jichangjichang) [inline clients/samples/gemm/sample_hipblaslt_gemm_dgelu_bgrad.cpp]: h_bias and hipMemcpy are not necessary

Comment (waynehuangntu) [reply clients/samples/gemm/sample_hipblaslt_gemm_dgelu_bgrad.cpp]: Done modification and validation.
Please take a look, thx!

**Comment by jayhawk-commits** [comment]
Please resolve merge conflicts or close this PR to complete the task of importing PRs from this repo to the monorepo.
